tests/benchmarks/test_snowflake.py

```diff
@@ -0,0 +1,58 @@
+import os
+import uuid
+
+import dask
+import dask.dataframe as dd
+import pytest
+from dask_snowflake import read_snowflake, to_snowflake
+from distributed import Client
+from snowflake.sqlalchemy import URL
+from sqlalchemy import create_engine
+
+
+@pytest.fixture
+def client():
+    with Client(n_workers=2, threads_per_worker=10) as client:
+        yield client
+
+
+@pytest.fixture
+def table(connection_kwargs):
+    name = f"test_table_{uuid.uuid4().hex}".upper()
+
+    yield name
+
+    engine = create_engine(URL(**connection_kwargs))
+    engine.execute(f"DROP TABLE IF EXISTS {name}")
+
+
+@pytest.fixture(scope="module")
+def connection_kwargs():
+    return dict(
+        user=os.environ["SNOWFLAKE_USER"],
+        password=os.environ["SNOWFLAKE_PASSWORD"],
+        account=os.environ["SNOWFLAKE_ACCOUNT"],
+        database="testdb",
+        schema="public",
+        warehouse=os.environ["SNOWFLAKE_WAREHOUSE"],
+        role=os.environ["SNOWFLAKE_ROLE"],
+    )
+
+
+ddf = dask.datasets.timeseries(
+    start="2000-01-01", end="2000-12-31", freq="1T", partition_freq="1W"
+)
+
+
+def test_write_read_roundtrip(table, connection_kwargs, small_client):
```

Comment:
This is more of a question, do we want the test to be a round-trip, or having separate tests for write and read, could give us more information about where things are going wrong?

Reply:
I don't want the test to be round trip and I agree that individual read/write benchmarks would be better.

My thinking is we should switch out the `small_client` fixture for `small_cluster` and then individually benchmark each step in a single test.

The reason for this is that writing a single test benchmark writing is easy enough, but then we'll need to clean up the test's side effects and remove that table. As a result, when we go to test _reading_ the table, we'll have to redundantly write the table yet again. There might be a clever way around this that I can't see. But individually benchmarking each step in a single test makes sense in my head.